### PR TITLE
Run a linter on the docs

### PR DESCRIPTION
**Pull request by hbokh**
rst-lint: https://github.com/twolfson/restructuredtext-lint

Fixes these warnings in CI:

- WARNING: Title underline too short.
- WARNING: Title overline too short.

Other fixes:

- Cannot analyze code. No Pygments lexer found for "none", for code blocks

### CODE REVIEW

File: docs/user-guide/wingen.rst [hunk 1]
```diff
@@ -116,41 +116,41 @@ Create a file /etc/tftpd.rules:
     rg	\\					/ # Convert backslashes to slashes
```
Comment (hbokh) [inline docs/user-guide/wingen.rst:116]: To me this is a very weird code block!!
See under https://cobbler.readthedocs.io/en/latest/user-guide/wingen.html#preparing-for-an-unattended-network-installation-of-windows

Is that how a file `/etc/tftpd.rules` is supposed to look? @tpw56j

**Review by hbokh — COMMENTED** [review]
Added a comment on the block for `/etc/tftpd.rules` in file `wingen.rst`.

**Review by hbokh — COMMENTED** [review]
More cleanup.

Compare this (current): https://cobbler.readthedocs.io/en/latest/user-guide/wingen.html

with this (cleanup): https://cobbler--2538.org.readthedocs.build/en/2538/user-guide/wingen.html

File: docs/user-guide/wingen.rst (hunk 1 above)
Comment (tpw56j) [reply docs/user-guide/wingen.rst:118]: @hbokh The manual does not contain the latest changes and I am going to rewrite it in the near future. 

The path to the BCD file is hardcoded in bootmgr.exe with backslashes, so you need a tftp rule to convert them to forward slashes. This is how this tftp rule 0 works:
```
Feb 03 16:32:45 cobbler in.tftpd[27413]: Client ::ffff:192.168.130.228 finished boot9ea.exe
Feb 03 16:32:45 cobbler in.tftpd[27414]: remap: input: \Boot\9ea
Feb 03 16:32:45 cobbler in.tftpd[27414]: remap: rule 0: rewrite: /Boot\9ea
Feb 03 16:32:45 cobbler in.tftpd[27414]: remap: rule 0: rewrite: /Boot/9ea
Feb 03 16:32:45 cobbler in.tftpd[27414]: remap: rule 19: rewrite: /images/Win2019-Server_EN-x64/9ea
Feb 03 16:32:45 cobbler in.tftpd[27414]: remap: done
```
How /etc/tftpd.rules will look depends on the filenames of the windows boot files that will be generated by the sync_post_wingen trigger.

For example:
```
# cobbler profile report --name=Win2019-Server_EN-x64
Name                           : Win2019-Server_EN-x64
Automatic Installation Template : win.ks
Automatic Installation Metadata : {'kernel': 'win9ea.0', 'bootmgr': 'boot9ea.exe', 'bcd': '9ea', 'winpe': 'winpe.wim', 'answerfile': 'autounattended.xml'}
```

/etc/tftpd.rules for profile Win2019-Server_EN-x64:
```
rg	\\					/ # Convert backslashes to slashes
r	(boot9e.\.exe)				/images/Win2019-Server_EN-x64/\1
r	(/Boot/)(9e.)				/images/Win2019-Server_EN-x64/\2
```
sync_post_wingen log:
```
Profile: Win2019-Server_EN-x64
Build answer file: /var/lib/tftpboot/images/Win2019-Server_EN-x64/autounattended.xml
running: /usr/bin/sha1sum /var/lib/tftpboot/images/Win2019-Server_EN-x64/autounattended.xml
received on stdout: 3efdd7cb609f32514b62878f9e14131db984888d  /var/lib/tftpboot/images/Win2019-Server_EN-x64/autounattended.xml

received on stderr: 
trying cachelink /var/lib/tftpboot/images/Win2019-Server_EN-x64/autounattended.xml -> /var/www/cobbler/distro_mirror/.link_cache/3efdd7cb609f32514b62878f9e14131db984888d -> /var/www/cobbler/distro_mirror/Win2019-Server_EN-x64/autounattended.xml
copying: /var/lib/tftpboot/images/Win2019-Server_EN-x64/autounattended.xml -> /var/www/cobbler/distro_mirror/Win2019-Server_EN-x64/autounattended.xml
trying symlink /var/lib/tftpboot/images/Win2019-Server_EN-x64/autounattended.xml -> /var/www/cobbler/images/Win2019-Server_EN-x64/autounattended.xml
Patching build Loader: /var/lib/tftpboot/images/Win2019-Server_EN-x64/boot9ea.exe
Build Loader: /var/lib/tftpboot/images/Win2019-Server_EN-x64/boot9ea.exe from /var/www/cobbler/distro_mirror/Win2019-Server_EN-x64/Boot/bootmgr.exe
trying symlink /var/lib/tftpboot/images/Win2019-Server_EN-x64/boot9ea.exe -> /var/www/cobbler/images/Win2019-Server_EN-x64/boot9ea.exe
Build PXEBoot: /var/lib/tftpboot/images/Win2019-Server_EN-x64/win9ea.0 from /var/www/cobbler/distro_mirror/Win2019-Server_EN-x64/Boot/pxeboot.n12
trying symlink /var/lib/tftpboot/images/Win2019-Server_EN-x64/win9ea.0 -> /var/www/cobbler/images/Win2019-Server_EN-x64/win9ea.0
Build BCD: /var/lib/tftpboot/images/Win2019-Server_EN-x64/9ea from /var/www/cobbler/distro_mirror/Win2019-Server_EN-x64/Boot/bcd for /images/Win2019-Server_EN-x64/winpe.wim
trying symlink /var/lib/tftpboot/images/Win2019-Server_EN-x64/9ea -> /var/www/cobbler/images/Win2019-Server_EN-x64/9ea
running: ['/usr/bin/cp', '--reflink=auto', '/var/www/cobbler/distro_mirror/Win2019-Server_EN-x64/Boot/winpe.wim', '/var/lib/tftpboot/images/Win2019-Server_EN-x64/winpe.wim']
received on stdout: 
received on stderr: 
trying symlink /var/lib/tftpboot/images/Win2019-Server_EN-x64/winpe.wim -> /var/www/cobbler/images/Win2019-Server_EN-x64/winpe.wim
running: ['/usr/bin/wimupdate', '/var/lib/tftpboot/images/Win2019-Server_EN-x64/winpe.wim', '--command=add /tmp/tmpubi9k6py /Windows/System32/startnet.cmd']
received on stdout: Scanning "/tmp/tmpubi9k6py" (loading as WIM path: "/Windows/System32/startnet.cmd")...

264 bytes scanned (1 files, 0 directories)    
264 bytes scanned (1 files, 0 directories)    
Updating "/Windows/System32/startnet.cmd" in WIM image
Using LZX compression with 1 thread

Archiving file data: 0 bytes of 264 bytes (0%) done
Archiving file data: 264 bytes of 264 bytes (100%) done
```

**Comment by SchoolGuy** [comment]
@hbokh Is it possible to let that run as a Github Action? If yes, I would be very happy if you could add it here!

**Comment by tpw56j** [comment]
@hbokh Many thanks! It looks much better that way. I am going to update this documentation and bring it in line with the latest changes in the code. May I hope for your help in preparing for the next revision of this document?

**Comment by hbokh** [comment]
> @hbokh May I hope for your help in preparing for the next revision of this document?

I can always review it. But Windows, so I don't really understand it.

**Comment by hbokh** [comment]
> @hbokh Is it possible to let that run as a Github Action? If yes, I would be very happy if you could add it here!

It would be possible, but currently it runs into 124 erros, most of them (103) are `Unknown directive type "automodule".`
Needs a workaround or a fix for that.

**Comment by hbokh** [comment]
Added GH Action to run a lint on the docs directory with`rstcheck`.
Error in the checks is the same when run on my local environment 🎉:

```
$ rstcheck -r docs
docs/user-guide/extending-cobbler.rst:132: (ERROR/3) (python) unindent does not match any outer indentation level
docs/user-guide/configuration-management-integrations.rst:101: (WARNING/2) (rst) Link is formatted in Markdown style.
docs/user-guide/web-interface.rst:47: (WARNING/2) (rst) Link is formatted in Markdown style.
```

I'm having a hard time fixing both the indent AND the links in MD.

**Review by hbokh — COMMENTED** [review]
Linter in GH Actions is now working.

**Review by tpw56j — APPROVED** [review]
@hbokh  I think the documentation is better now

**Review by SchoolGuy — APPROVED** [review]
LGTM. This is huge! Thanks for this @hbokh !

**Comment by SchoolGuy** [comment]
Since we have two approvals (people not part of the org don't count officially), I will merge this now!

**Pull request merged**